apps/admin_app/assets/js/components/CreatePromotions.tsx

```diff
@@ -0,0 +1,559 @@
+            }).catch(error => console.error('Error', error))
+        }
+    }
+
+    onSuccess = () => {
+        this.setState({ success: true })
+    }
+    onFail = (response) => {
+        this.setState({ failed: true, errors: response })
+        Object.keys(response).map((res) => {
+            var errormsg = ""
+            Object.keys(response[res]).map((res1) => {
+                errormsg = errormsg + res1 + " " + response[res][res1][0]["message"] + "\n"
+            })
+        })
+    }
+    handleAddRule = () => {
+        this.setState({ addRule: true })
+        const url = myconstants.RULES_LIST_URL;
+        fetchGet(url).then(res => res.json()).then(response => {
+            var availableRules = response
+            this.setState({ availableRules: availableRules })
+        })
+    }
+
+    handleAddAction = () => {
+        this.setState(prevState => ({
+            actionDown: !prevState.actionDown,
+            addAction: !prevState.addAction
+        }))
+
+        const url = myconstants.ACTIONS_LIST_URL;
+        fetchGet(url).then(res => res.json()).then(response => {
+            var availableActions = response
+            this.setState({ availableActions: availableActions })
+        })
+            .catch(error => console.error('Error', error));
+
+        fetchGet(myconstants.CALCULATORS_LIST_URL).then(res => res.json()).then(response => {
+            var availableCalculators = response
+            this.setState({ availableCalculators: availableCalculators })
+        })
+            .catch(error => console.error('Error', error));
+    }
+
+    handleRule = (rule) => {
+        this.setState({ selectedRule: rule })
+        const url = myconstants.RULE_PREFERENCES_URL
+        var data = { "rule": rule }
+        fetchPost(url, data).then(res => res.json()).then(response => { }).catch(error => console.error('Error', error));
+    }
+
+    handleSaveRule = (rule) => {
+        const addedRules = this.state.addedRules
+        addedRules.push(rule)
+        this.setState({ addedRules: addedRules })
+    }
+
+    handleSaveAction = (action) => {
+
+        const addedActions = this.state.addedActions
+        addedActions.push(action)
+        this.setState({ addedActions: addedActions })
+    }
+
+    handleDeleteRule = (index) => {
+        var addedRules = this.state.addedRules;
+        addedRules.splice(index, 1)
+        this.setState({ addedRules: addedRules })
+    }
+
+    handleDeleteAction = (index) => {
+        var addedActions = this.state.addedActions;
+        addedActions.splice(index, 1)
+        this.setState({ addedActions: addedActions })
+    }
+
+    handleProducts = (products) => {
+        const product_list = []
+        products.map(item => { product_list.push(parseInt(item)) })
+        this.setState({ selectedProducts: product_list })
+    }
+
+    renderRuleOptions = () => {
+        switch (this.state.selectedRule) {
+            case myconstants.ORDER_TOTAL_MODULE:
```

Comment:
change variable name of "myconstants"

Reply:
@ashish173 PromotionsConstants is okay?

Reply:
If it is going to store just the promotion constants then I am fine with it. If it can store any other constants in that case we need to look for better names. @SakinaBoriwala21